Source/Core/DolphinQt/EditTimePlayedDialog.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "DolphinQt/EditTimePlayedDialog.h"
+
+#include <chrono>
+#include <string>
+
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QFormLayout>
+#include <QIntValidator>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QString>
+#include <QWidget>
+
+#include "Core/TimePlayed.h"
+#include "Core/TitleDatabase.h"
+
+#include "UICommon/GameFile.h"
+
+EditTimePlayedDialog::EditTimePlayedDialog(QWidget* const parent, const UICommon::GameFile& game)
+    : QDialog(parent), m_game_id(game.GetGameID())
+{
+  using namespace std::chrono_literals;
+
+  // std::chrono::milliseconds is guaranteed to have a signed integer representation type with at
+  // least 45 bits (44 non-sign bits). (2^44)/(3'600'000 milliseconds/hour) ~= 4'886'718 hours.
+  // Setting MAX_EDITABLE_HOURS to 999'999 makes all <= 6 digit hour counts valid and corresponds to
+  // 114 years, which is more than anyone will actually ever play.
+  constexpr auto MAX_EDITABLE_HOURS = std::chrono::hours(999'999);
+  constexpr std::chrono::milliseconds MAX_EDITABLE_TIME =
+      MAX_EDITABLE_HOURS + std::chrono::minutes(59);
+
+  const std::chrono::milliseconds time_played =
+      std::min(MAX_EDITABLE_TIME, TimePlayedManager::GetInstance().GetTimePlayed(m_game_id));
+  const auto hours_played = time_played / 1h;
+  const auto minutes_played = (time_played % 1h) / 1min;
+
+  setWindowTitle(QStringLiteral("Edit Time Played"));
+
+  auto* const layout = new QFormLayout;
+
+  m_hours_edit = new QLineEdit;
+  const auto hours_validator = new QIntValidator(0, MAX_EDITABLE_HOURS.count(), this);
+  m_hours_edit->setValidator(hours_validator);
+  m_hours_edit->setText(QString::number(hours_played));
+
+  m_minutes_edit = new QLineEdit;
+  const auto minutes_validator = new QIntValidator(0, 59, this);
+  m_minutes_edit->setValidator(minutes_validator);
+  m_minutes_edit->setText(QString::number(minutes_played));
+
+  m_invalid_minutes_label = new QLabel(tr("Minutes must be 0-59."));
+  m_invalid_minutes_label->hide();
+
+  auto* const button_box = new QDialogButtonBox(QDialogButtonBox::Cancel | QDialogButtonBox::Save);
+
+  const Core::TitleDatabase title_db;
+  layout->addRow(tr("Game:"), new QLabel(QString::fromStdString(game.GetName(title_db))));
+  layout->addRow(tr("ID:"), new QLabel(QString::fromStdString(m_game_id)));
+  layout->addRow(tr("Hours:"), m_hours_edit);
+  layout->addRow(tr("Minutes:"), m_minutes_edit);
+  layout->addWidget(m_invalid_minutes_label);
+  layout->addWidget(button_box);
```

Comment:
The button box doesn't follow the bottom edge when resizing the window.

Easiest way to fix this might be adding a row with a `QVBoxLayout` which has an `addStretch`.

Reply:
Given the constraints on the hour/minute inputs there's no benefit in expanding the window, so I've just made it non-resizable.